### DISCUSSION TITLE
Microsoft provides both powershell and terminal in windows 11 so is there any major difference between them?

### DISCUSSION THREAD

**Discussion by Greatz08** [PowerShell/PowerShell#23706] [General]
Both have similar usecase according to me so what are major differences between them usecase and if no major difference then why are there two almost same thing in one os?

**Comment by HotCakeX** [comment]
Windows Terminal is a terminal, it's not a scripting engine. Windows Terminal is the gateway to PowerShell, CMD, Azure cloud shell, WSL subsystem etc.

So they aren't even comparable.

**Reply by bondido** [reply]
Starting "Terminal" in my windows 11 starts PowerShell, so where is the point in saying "they aren't even comparable"?

You can run AZ CLI, WSL or even 'pwsh' :-) from powershell as well as from the Command Prompt.

**Reply by SeeminglyScience** [reply]
You can run other shell applications from a shell application, but they're all still hosted in the terminal. You can't for instance run host terminal/xterm.js/cmder/hyper inside pwsh itself, that wouldn't make any sense. The terminal is the thing that renders what the shell application is doing.

It's sort of like trying to compare Windows to Excel

**Comment by rhubarb-geek-nz** [comment]
> Starting "Terminal" in my windows 11 starts PowerShell, so where is the point in saying "they aren't even comparable"?

Terminal is a GUI application that emulates a traditional terminal in a Window. All it can do is print text and accept key presses.  It has no behaviour in itself over what should be printed in the window. What it does is run console applications as child processes to provide that, eg PowerShell, the Command prompt or anything else you choose to select or configure.

PowerShell is a console application (eg character mode app, not GUI), that accepts input in the form of key presses and prints output. It can be hosted by Windows Terminal, by the original classic console window, launched by SSHD or run with no console at all. PowerShell itself does not need to concern itself over moving windows etc, it gets on with the job of running scripts and commands, and responding to the users input. PowerShell can be run on a computer with no GUI at all.

So they are completely different things but work well together, and a common configuration is for Windows Terminal to default to running PowerShell to provide the content for the window.
